include/seqan3/alignment/pairwise/policy/alignment_matrix_policy.hpp

```diff
@@ -0,0 +1,146 @@
+    constexpr alignment_matrix_policy & operator=(alignment_matrix_policy const &) = default; //!< Defaulted.
+    constexpr alignment_matrix_policy & operator=(alignment_matrix_policy &&) = default; //!< Defaulted.
+    ~alignment_matrix_policy() = default; //!< Defaulted.
+    //!}
+
+    /*!\brief Allocates the memory of the underlying matrices.
+     * \param[in] fst_range The first range (corresponds to horizontal dimension).
+     * \param[in] snd_range The second range (corresponds to vertical dimension).
+     *
+     * \details
+     *
+     * Initialises the underlying score and trace matrices and sets the respective matrix iterators to the begin of the
+     * corresponding matrix.
+     */
+    template <typename fst_range_t, typename snd_range_t>
+    constexpr void allocate_matrix(fst_range_t && fst_range, snd_range_t && snd_range)
+    {
+        score_matrix = score_matrix_type{fst_range, snd_range};
+        trace_matrix = trace_matrix_type{fst_range, snd_range};
+
+        initialise_matrix_iterator();
+    }
+
+    /*!\brief Allocates the memory of the underlying matrices.
+     * \param[in] fst_range The first range (corresponds to horizontal dimension).
+     * \param[in] snd_range The second range (corresponds to vertical dimension).
+     * \param[in] band The band used to initialise the matrices.
+     * \param[in] cache The current cache used by the alignment algorithm.
+     *
+     * \details
+     *
+     * Initialises the underlying banded score and trace matrices and sets the respective matrix iterators to the begin
+     * of the corresponding matrix. Using the additional band parameter the actual dimensions are reduced according
+     * to the matrix implementation. For the banded case one additional cell per column is stored such that we can read
+     * from it without introducing a case distinction inside of the algorithm implementation. However, this cell needs
+     * to be properly initialised with an infinity value. To emulate the infinity for integral values we use the
+     * smallest representable value and subtract the gap extension score (assumed to be always negative) from it.
+     * In the algorithm we never write to this cell and only add the extension costs to the read value. This way we
+     * can get the smallest possible value as an infinity.
+     */
+    template <std::ranges::forward_range fst_range_t, std::ranges::forward_range snd_range_t, typename score_t>
+    constexpr void allocate_matrix(fst_range_t && fst_range,
+                                   snd_range_t && snd_range,
+                                   static_band const & band,
+                                   alignment_algorithm_cache<score_t> const & cache)
+    {
+        assert(cache.gap_extension_score <= 0); // We expect it to be always non-positive.
+
+        score_t inf = std::numeric_limits<score_t>::min() - cache.gap_extension_score;
```

Comment:
No I want this to be -inf if possible. The problem is that integers can't handle -inf, so this will use as much as possible from the score range for the alignment.

Reply:
I don't get it. What is wrong with lowest?